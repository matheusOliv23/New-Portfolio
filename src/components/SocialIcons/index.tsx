import { Icon } from "@iconify/react"

const icons = [
  {
    id: 1,
    title: "Instagram",
    name: "ph:instagram-logo",
    href: "https://www.instagram.com/matheus.olli23/"
  },
  {
    id: 2,
    title: "Linkedin",
    name: "mdi:linkedin",
    href: "https://www.linkedin.com/in/matheushenrioliveira/"
  },
  {
    id: 3,
    title: "Github",
    name: "mdi:github",
    href: "https://github.com/matheusOliv23"
  }
]

export default function SocialIcons() {
  return (
    <div className="flex gap-5 mt-2">
      {icons.map((icon) => (
        <div
          className="flex hover:opacity-50 flex-col items-center justify-center"
          key={icon.name}
        >
          <a href={icon.href} target="_blank" rel="noreferrer">
            <Icon
              aria-label={icon.name}
              width="32"
              height="32"
              icon={icon.name}
            />
          </a>
          <span className="text-xs">{icon.title}</span>
        </div>
      ))}
    </div>
  )
}
