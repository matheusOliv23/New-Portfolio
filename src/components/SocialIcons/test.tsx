import { render, screen } from "@testing-library/react"

import SocialIcons from "./index"

const { getByText } = screen

describe("<SocialIcons />", () => {
  it("should render by default", () => {
    render(<SocialIcons />)

    const instagram = getByText("Instagram")
    const linkedin = getByText("Linkedin")
    const github = getByText("Github")

    expect(instagram).toBeInTheDocument()
    expect(linkedin).toBeInTheDocument()
    expect(github).toBeInTheDocument()
  })

  it("should match snapshot", () => {
    const { container } = render(<SocialIcons />)

    expect(container).toMatchSnapshot()
  })
})
