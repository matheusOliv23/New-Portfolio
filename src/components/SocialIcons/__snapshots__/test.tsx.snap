// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialIcons /> should match snapshot 1`] = `
<div>
  <div
    class="flex gap-5 mt-2"
  >
    <div
      class="flex hover:opacity-50 flex-col items-center justify-center"
    >
      <a
        href="https://www.instagram.com/matheus.olli23/"
        rel="noreferrer"
        target="_blank"
      >
        <span />
      </a>
      <span
        class="text-xs"
      >
        Instagram
      </span>
    </div>
    <div
      class="flex hover:opacity-50 flex-col items-center justify-center"
    >
      <a
        href="https://www.linkedin.com/in/matheushenrioliveira/"
        rel="noreferrer"
        target="_blank"
      >
        <span />
      </a>
      <span
        class="text-xs"
      >
        Linkedin
      </span>
    </div>
    <div
      class="flex hover:opacity-50 flex-col items-center justify-center"
    >
      <a
        href="https://github.com/matheusOliv23"
        rel="noreferrer"
        target="_blank"
      >
        <span />
      </a>
      <span
        class="text-xs"
      >
        Github
      </span>
    </div>
  </div>
</div>
`;
